src/cmap/wire_protocol/on_demand/string_finder.ts

```diff
@@ -0,0 +1,23 @@
+import { ByteUtils } from '../../../utils';
+
+/**
+ * @internal
+ * StringFinder is declared as a class so we can delete/test the cache
+ */
+export class StringFinder {
+  private static cache: Record<string, Uint8Array> = Object.create(null);
```

Comment:
Since our intention for now is to only use OnDemand internally this should only grow to be as large as the keys depended on by the driver (numbering in the 100s at absolute most). If this were made public we would need to think about eviction.